packages/core/test/skip_logic.ts

```diff
@@ -22,7 +33,66 @@ describe("Skip logic", () => {
     const reg = new Registry().addFile(file);
     await reg.parseAsync();
     expect(reg.getObjects().length).to.equal(1);
-    const skip = new SkipLogic(reg);
-    expect(skip.isGeneratedGatewayClass(reg.getObjects()[0] as Class)).to.equal(true);
+    expect(new SkipLogic(reg).skip(reg.getObjects()[0])).to.equal(true);
+  });
+
+  it("interface, isGeneratedProxy", async () => {
+    const abap = "INTERFACE zif_foobar PUBLIC.\n" +
+      "  METHODS method1 IMPORTING foo TYPE i.\n" +
+      "ENDINTERFACE.";
+    const reg = new Registry().addFile(new MemoryFile("zif_foobar.intf.abap", abap));
+
+    const xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+      "<abapGit version=\"v1.0.0\" serializer=\"LCL_OBJECT_INTF\" serializer_version=\"v1.0.0\">\n" +
+      "<asx:abap xmlns:asx=\"http://www.sap.com/abapxml\" version=\"1.0\">\n" +
+      "<asx:values>\n" +
+      "<VSEOINTERF>\n" +
+      "<CLSNAME>ZIF_FOOBAR</CLSNAME>\n" +
+      "<LANGU>E</LANGU>\n" +
+      "<DESCRIPT>Proxy Interface (generated)</DESCRIPT>\n" +
+      "<EXPOSURE>2</EXPOSURE>\n" +
+      "<STATE>1</STATE>\n" +
+      "<UNICODE>X</UNICODE>\n" +
+      "<CLSPROXY>X</CLSPROXY>\n" +
+      "</VSEOINTERF>\n" +
+      "</asx:values>\n" +
+      "</asx:abap>\n" +
+      "</abapGit>";
+    reg.addFile(new MemoryFile("zif_foobar.intf.xml", xml));
+
+    await reg.parseAsync();
+    expect(reg.getObjects().length).to.equal(1);
+    expect(new SkipLogic(reg).skip(reg.getObjects()[0])).to.equal(true);
+  });
+
+  it("generated BOPF constants interface", async () => {
+    const abap = `
+INTERFACE zif_tt_i_projects_c PUBLIC.
```

Comment:
This is a name for the generated interface, each BOPF generates its own interface with different names

Reply:
yea, this is just the unit test

the logic checks for the interface